cpp/daal/src/services/service_topo.cpp

```diff
@@ -2111,20 +2111,20 @@ namespace services
 {
 namespace internal
 {
-unsigned getNumElementsFitInMemory(size_t sizeofMemory, size_t sizeofAnElement, size_t defaultNumElements)
+size_t getNumElementsFitInMemory(size_t sizeofMemory, size_t sizeofAnElement, size_t defaultNumElements)
 {
     const size_t n = sizeofMemory / sizeofAnElement;
     if (n < 1) return sizeofMemory ? 1 : defaultNumElements;
```

Comment:
Let's face it - if an element doesn't fit to memory - it can be one of 2 options:
1) An element size have been computed incorrectly. Most likely the integer underflow happened at some point
2) Size of memory value have been miscalculated or corrupted

 In any case I would prefer to raise an exception here

Reply:
We do not return service::status from this function and throwing exception is not a common pattern in DAAL library. That is why I suggest that we just add DAAL_ASSERT above this if-statement without removing it.

Reply:
In fact, for now we can mark this place as TODO in order not to break functionality and investigate if we can make this update after release.